crates/syntax/src/ast/make.rs

```diff
@@ -596,10 +597,17 @@ pub fn fn_(
         Some(it) => format!("{} ", it),
     };
 
-    ast_from_text(&format!(
-        "{}fn {}{}{} {}{}",
-        visibility, fn_name, type_params, params, ret_type, body
-    ))
+    if is_async {
+        ast_from_text(&format!(
+            "{}async fn {}{}{} {}{}",
+            visibility, fn_name, type_params, params, ret_type, body
+        ))
+    } else {
+        ast_from_text(&format!(
+            "{}fn {}{}{} {}{}",
+            visibility, fn_name, type_params, params, ret_type, body
+        ))
+    }
```

Comment:
Instead of duplicating we can have a a variable thats either `""` or `"async"` depending on is_async and then format that into the string here